src/resty/ljsonschema/init.lua

```
@@ -1023,25 +1059,32 @@ generate_validator = function(ctx, schema)
       local validator = ctx:validator({ 'oneOf', tostring(i-1) }, subschema)
       ctx:stmt(sformat('  if %s(%s) then', validator, ctx:param(1)))
       ctx:stmt(        '    if matched then')
-      ctx:stmt(sformat('      return false, %s("value should match only one schema, but matches both schemas %%d and %%d", matched, %d)',
+      ctx:handle_error('      ', sformat('%s("value should match only one schema, but matches both schemas %%d and %%d", matched, %d)',
                        ctx:libfunc('string.format'), i))
       ctx:stmt(        '    end')
       ctx:stmt(        '    matched = ', tostring(i))
       ctx:stmt(        '  end')
     end
     ctx:stmt('  if not matched then')
-    ctx:stmt('    return false, "value should match only one schema, but matches none"')
+    ctx:handle_error('    ', '"value should match only one schema, but matches none"')
     ctx:stmt('  end')
     ctx:stmt('end')
   end
 
   if schema['not'] then
     local validator = ctx:validator({ 'not' }, schema['not'])
     ctx:stmt(sformat('if %s(%s) then', validator, ctx:param(1)))
-    ctx:stmt(        '  return false, "value wasn\'t supposed to match schema"')
+    ctx:handle_error('  ', '"value wasn\'t supposed to match schema"')
     ctx:stmt(        'end')
   end
 
+  if ctx._root._collect_all_errors then
+    ctx:stmt('if #errors > 0 then')
+    ctx:stmt('  local err = table.concat(errors, "; ")')
+    ctx:stmt('  return false, err')
```

Why not return the error table directly, like:
`return false, "validate failed", errors`

---

Referring to [the jsonschema repo](https://docs.rs/jsonschema/0.33.0/jsonschema/index.html),  I'm trying to make an error contain schema_path(Path to the JSON Schema keyword that failed validation) and error_msg.